addons/website_hr_recruitment/controllers/main.py

```diff
@@ -3,74 +3,190 @@
 
 from odoo import http, _
 from odoo.addons.http_routing.models.ir_http import slug
+from odoo.osv.expression import AND
 from odoo.http import request
+from odoo.tools.misc import groupby
 from werkzeug.exceptions import NotFound
 
 
 class WebsiteHrRecruitment(http.Controller):
+    _jobs_per_page = 12
+
     def sitemap_jobs(env, rule, qs):
         if not qs or qs.lower() in '/jobs':
             yield {'loc': '/jobs'}
 
     @http.route([
         '/jobs',
-        '/jobs/country/<model("res.country"):country>',
-        '/jobs/department/<model("hr.department"):department>',
-        '/jobs/country/<model("res.country"):country>/department/<model("hr.department"):department>',
-        '/jobs/office/<int:office_id>',
-        '/jobs/country/<model("res.country"):country>/office/<int:office_id>',
-        '/jobs/department/<model("hr.department"):department>/office/<int:office_id>',
-        '/jobs/country/<model("res.country"):country>/department/<model("hr.department"):department>/office/<int:office_id>',
-        '/jobs/employment_type/<int:contract_type_id>',
-        '/jobs/country/<model("res.country"):country>/employment_type/<int:contract_type_id>',
-        '/jobs/department/<model("hr.department"):department>/employment_type/<int:contract_type_id>',
-        '/jobs/office/<int:office_id>/employment_type/<int:contract_type_id>',
-        '/jobs/country/<model("res.country"):country>/department/<model("hr.department"):department>/employment_type/<int:contract_type_id>',
-        '/jobs/country/<model("res.country"):country>/office/<int:office_id>/employment_type/<int:contract_type_id>',
-        '/jobs/department/<model("hr.department"):department>/office/<int:office_id>/employment_type/<int:contract_type_id>',
-        '/jobs/country/<model("res.country"):country>/department/<model("hr.department"):department>/office/<int:office_id>/employment_type/<int:contract_type_id>',
```

Comment:
quid about retrocompatibility for old links?
Some very specific can probably just redirect to more global page.
But we don't want a 404 for each these rules e.g. /jobs/country/Belgium-20

Users work for seo, it is not to be broken after each upgrade.
![image](https://user-images.githubusercontent.com/4848797/191279610-adfc71ee-f569-45e3-8762-91c0311606de.png)

Reply:
Indeed. We have adapted the routes in order to maintain backward compatibility.

The changes are included in the commit below:

https://github.com/odoo/odoo/pull/67913/commits/bdd855924ba4faef07f48b4b4c0fe6790faa90ee

Thanks!

@bso-odoo